src/randoop/operation/MethodCall.java

```diff
@@ -340,13 +335,74 @@ public boolean isStatic() {
    * java.util.ArrayList.get(int)
    * java.util.ArrayList.add(int,java.lang.Object)
    */
+  @Override
   public String toParseableString() {
-    return Reflection.getSignature(method);
+    return this.getSignature();
   }
 
-  public static StatementKind parse(String s) {
-    return RMethod.getRMethod(Reflection.getMethodForSignature(s));
+  public String getSignature() {
+    StringBuilder sb = new StringBuilder();
+    sb.append(method.getDeclaringClass().getName() + ".");
+    sb.append(method.getName() + "(");
+    Class<?>[] params = method.getParameterTypes();
+    for (int j = 0; j < params.length; j++) {
+      sb.append(params[j].getName());
+      if (j < (params.length - 1))
+        sb.append(",");
+    }
+    sb.append(")");
+    return sb.toString();
+  }
+
+  public static Operation parse(String s) throws OperationParseException {
+    return MethodCall.getMethodCall(MethodParser.getMethodForSignature(s));
+  }
+
+  /**
+   * getDeclaringClass returns the class in which this method is declared.
+   */
+  @Override
+  public Class<?> getDeclaringClass() {
+    return method.getDeclaringClass();
+  }
+
+  /**
+   * isMethodIn determines whether the current method object
+   * calls one of the methods in the list.
+   * @param list method objects to compare against.
+   * @return true if method called by this object is in the given list.
+   */
+  public boolean isMethodIn(List<Method> list) {
+    return list != null && list.contains(method);
+  }
+
+  /**
+   * callsTheMethod determines whether the method that this object calls is 
+   * method given in the parameter.
+   * @param m method to test against.
+   * @return true, if m corresponds to the method in this object, false, otherwise.
+   */
+  public boolean callsTheMethod(Method m) {
+    return method.equals(m);
   }
 
+  /**
+   * isMessage is a predicate to indicate that this method call is a message to a receiver object.
```

Comment:
Can you clarify what is a message to a receiver object?  What wouldn't be one?
Should this use `@inheritDoc` instead?